src/kernels/MIOpenLossSum.cpp

```diff
@@ -0,0 +1,69 @@
+#ifndef MIOPEN_DONT_USE_HIP_RUNTIME_HEADERS
+#include <hip/hip_fp16.h>
+#include <hip/hip_runtime.h>
+#endif
+
+#include "float_types.h"
+
+#ifndef REDUCE_SIZE
+#define REDUCE_SIZE 256
+#endif
+
+#ifndef IN_OUT_TYPE
+#define IN_OUT_TYPE float
+#endif
+
+__device__ FLOAT_ACCUM warp_reduce_sum(FLOAT_ACCUM val)
```

Comment:
Please separate warp_reduce_sum and block_reduce_sum to the header file.

Reply:
I have separated these func to warp_shuffle.hpp